File: DroidTalksAndroid/app/src/main/java/com/arturomejiamarmol/droidtalks/topics/TopicsAdapter.java
package com.arturomejiamarmol.droidtalks.topics;

import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.arturomejiamarmol.droidtalks.R;


import java.util.List;

import com.arturomejiamarmol.droidtalks.ui.tools.Utils;
public class TopicsAdapter extends RecyclerView.Adapter<TopicsAdapter.ViewHolder> {
    

    private final OnListFragmentInteractionListener mListener;

    private List<String> topics;

    public TopicsAdapter(List<String> topics, OnListFragmentInteractionListener listener) {
        mListener = listener;
        this.topics = topics;

    }



    @Override
    public ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.topics_item_list, parent, false);
        return new ViewHolder(view);
    }
    
    @Override
    public void onBindViewHolder(final ViewHolder holder, int position) {
        holder.mTopic.setText(topics.get(position));
        holder.mTopic.setTextColor(Utils.getMatColor("500",holder.mTopic.getContext()));

        holder.mView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                if (null != mListener) {
                    // Notify the active callbacks interface (the activity, if the
                    // fragment is attached to one) that an item has been selected.
                    mListener.onListFragmentInteraction(holder.mTopic.getText().toString());
                }
            }
        });
    }
    
    @Override
    public int getItemCount() {
        return topics.size();
    }
    
    public class ViewHolder extends RecyclerView.ViewHolder {
        public final View mView;
        public final TextView mTopic;


        public ViewHolder(View view) {
            super(view);
            mView = view;
            mTopic = (TextView) view.findViewById(R.id.topic);
        }
        
        @Override
        public String toString() {
            return super.toString() + " '" ;}
    }
}

File: DroidTalksAndroid/app/src/main/java/com/arturomejiamarmol/droidtalks/topics/TopicsAdapter.kt
package com.arturomejiamarmol.droidtalks.topics

import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView

import com.arturomejiamarmol.droidtalks.R
import com.arturomejiamarmol.droidtalks.data.Topic
import com.arturomejiamarmol.droidtalks.ui.tools.*

class TopicsAdapter(private val topics: List<Topic>, private val mListener: OnListFragmentInteractionListener) : RecyclerView.Adapter<TopicsAdapter.ViewHolder>() {


    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view = LayoutInflater.from(parent.context)
                .inflate(R.layout.topics_item_list, parent, false)
        return ViewHolder(view)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val topic = topics[position]

        holder.mTopic.text = topic.name
        holder.mCount.text = if (topic.resourcesCount >= 100) "99+" else topic.resourcesCount.toString()

        holder.mCount.setBackgroundColor(getMatColor("500", holder.mTopic.context))

        holder.mView.setOnClickListener {
            mListener.onListFragmentInteraction(holder.mTopic.text.toString())
        }
    }

    override fun getItemCount(): Int = topics.size

    inner class ViewHolder(val mView: View) : RecyclerView.ViewHolder(mView) {
        val mTopic: TextView = mView.findViewById<TextView>(R.id.topic)
        val mCount: TextView = mView.findViewById<TextView>(R.id.count)
    }
}
